aidial_adapter_bedrock/llm/converse/input.py

```diff
@@ -0,0 +1,293 @@
+
+    if message.role == DialRole.FUNCTION:
+        return [function_result_to_content_part(message)]
+    elif message.role == DialRole.TOOL:
+        return [tool_result_to_content_part(message)]
+
+    content = []
+    match message.content:
+        case str():
+            content.append({"text": message.content})
+        case list():
+            for part in message.content:
+                match part:
+                    case MessageContentTextPart():
+                        content.append({"text": part.text})
+                    case MessageContentImagePart():
+                        resource = await URLResource(
+                            url=part.image_url.url,
+                            supported_types=supported_image_types,
+                        ).download(storage)
+                        content.append(
+                            {
+                                "image": {
+                                    "format": to_converse_image_type(
+                                        resource.type
+                                    ),
+                                    "source": {
+                                        "bytes": resource.data,
+                                    },
+                                }
+                            }
+                        )
+        case None:
+            pass
+        case _:
+            assert_never(message.content)
+
+    if message.custom_content and message.custom_content.attachments:
+        for attachment in message.custom_content.attachments:
+            resource = await AttachmentResource(attachment=attachment).download(
+                storage
+            )
+            content.append(
+                {
+                    "image": {
+                        "format": to_converse_image_type(resource.type),
+                        "source": {"bytes": resource.data},
+                    }
+                }
+            )
+    if message.function_call and message.tool_calls:
+        raise ValidationError(
+            "You cannot use both function call and tool calls in the same message"
+        )
+    elif message.function_call:
+        content.append(function_call_to_content_part(message.function_call))
+    elif message.tool_calls:
+        content.extend(
+            [
+                tool_call_to_content_part(tool_call)
+                for tool_call in message.tool_calls
+            ]
+        )
+
+    return content
+
+
+async def to_converse_message(
+    message: DialMessage,
+    storage: FileStorage | None,
+    supported_image_types: list[str] | None = None,
+) -> ConverseMessage:
+
+    return {
+        "role": to_converse_role(message.role),
+        "content": await _get_converse_message_content(
+            message, storage, supported_image_types
+        ),
+    }
+
+
+def get_converse_system_prompt(
+    messages: List[DialMessage],
+) -> ConverseTextPart | None:
+    system_msgs = []
+    saw_non_system = False
```

Comment:
`found_non_system` is a more common naming for such flags.

Reply:
Fixed